common/interface/common.ts

```diff
@@ -12,3 +19,151 @@ export type TErrorResponse = {
 export type TAction<TRequest, TResponse> =
   (req: TRequest) => Promise<TResponse>
   ;
+
+export type TTag = {
+  id: string,
+  name: string,
+};
+
+/*export type TDate = {
+  year: number,
+  month: number,
+  day: number,
+  hour: number,
+  minute: number,
+};*/
+export type TDate = Date;
+
+export type TPlace = {
+  name: string,
+  latitude: number,
+  longitude: number,
+};
+
+export type TPhoto = {
+  link: string,
+};
+
+export type TCommonData = {
+  id: string,
+  userId: string,
+  name: string,
+  description: string,
+  place?: TPlace,
+  photo?: TPhoto,
+  tags?: TTag[],
+  create: TDate,
+};
+
+export type TCommonDataNew = {
+  userId: string,
+  name: string,
+  description: string,
+  place?: TPlace,
+  photo?: TPhoto,
+  tags?: TTag[],
+};
+
+// Common Actions
+export type TGenericResponse = {
+  wasError: boolean,
```

Comment:
redundant. remove this. We know, that response resulted in error because error isn't undefined